File: core/src/main/java/fr/sncf/osrd/infra_state/api/SignalizationState.java
package fr.sncf.osrd.infra_state.api;

import fr.sncf.osrd.infra.api.reservation.ReservationRoute;
import fr.sncf.osrd.infra.api.signaling.Signal;
import java.util.Set;

public interface SignalizationState extends SignalizationStateView {

    /** Notify that the route state has changed. Returns a set of updated signals */
    Set<Signal<?>> notifyUpdate(ReservationRoute route);
}

File: core/src/main/kotlin/fr/sncf/osrd/dyn_infra/api/SignalizationState.kt
package fr.sncf.osrd.dyn_infra.api

import fr.sncf.osrd.infra.api.reservation.ReservationRoute
import fr.sncf.osrd.infra.api.signaling.Signal

interface SignalizationState : SignalizationStateView {
    /** Notify that the route state has changed. Returns a set of updated signals  */
    fun notifyUpdate(route: ReservationRoute): Set<Signal<*>>
}